spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesDiscoveryClientUtils.java

```diff
@@ -101,4 +107,29 @@ static Map<String, String> serviceMetadata(KubernetesDiscoveryProperties propert
 		return serviceMetadata;
 	}
 
+	static void postConstruct(List<SharedInformerFactory> sharedInformerFactories,
+			KubernetesDiscoveryProperties properties, Supplier<Boolean> informersReadyFunc,
+			List<Lister<V1Service>> serviceListers) {
+
+		sharedInformerFactories.forEach(SharedInformerFactory::startAllRegisteredInformers);
+		if (!Wait.poll(Duration.ofSeconds(1), Duration.ofSeconds(properties.cacheLoadingTimeoutSeconds()), () -> {
+			LOG.info(() -> "Waiting for the cache of informers to be fully loaded..");
+			return informersReadyFunc.get();
+		})) {
+			if (properties.waitCacheReady()) {
+				throw new IllegalStateException(
+					"Timeout waiting for informers cache to be ready, is the kubernetes service up?");
+			}
+			else {
+				LOG.warn(() -> "Timeout waiting for informers cache to be ready, " +
+					"ignoring the failure because waitForInformerCacheReady property is false");
+			}
+		}
+		else {
+			LOG.info(() -> "Cache fully loaded (total " + serviceListers.stream().mapToLong(x -> x.list().size()).sum()
```

Comment:
moved this log to the `else` branch, because if we hit a timeout, we were still logging it, kind of confusing, imo